helix-term/src/ui/editor.rs

```diff
@@ -275,9 +275,14 @@ impl EditorView {
                 let end = text.line_to_char(last_line + 1);
                 Range::new(start, end)
             };
-            let cursor_style = Style::default()
-                // .bg(Color::Rgb(255, 255, 255))
-                .add_modifier(Modifier::REVERSED);
+            let scope = match doc.mode() {
+                Mode::Insert => "ui.cursor.ins",
+                Mode::Select => "ui.cursor.sel",
+                Mode::Normal => "ui.cursor",
+            };
+            let cursor_style = theme
+                .try_get(scope)
+                .unwrap_or_else(|| Style::default().add_modifier(Modifier::REVERSED));
```

Comment:
Let's make it fallback to "ui.cursor" maybe and set the default for that to reversed?

Reply:
This works too though.

Reply:
Another thing I found cool: being able to style primary selection differently from the rest: https://github.com/Anomalocaridid/kakoune-cyberpunk-neon

Reply:
Added this too, seems like something worthwile to be able to distinguish the selection.